metaflow/plugins/argo/argo_workflows.py

```diff
@@ -0,0 +1,1304 @@
+                    + [
+                        "init",
+                        "--run-id %s" % run_id,
+                        "--task-id %s" % task_id_params,
+                    ]
+                    + [
+                        # Parameter names can be hyphenated, hence we use
+                        # {{foo.bar['param_name']}}
+                        "--%s={{workflow.parameters.%s}}"
+                        % (parameter["name"], parameter["name"])
+                        for parameter in self.parameters
+                    ]
+                )
+                if self.tags:
+                    init.extend("--tag %s" % tag for tag in self.tags)
+                # if the start step gets retried, we must be careful
+                # not to regenerate multiple parameters tasks. Hence
+                # we check first if _parameters exists already
+                exists = entrypoint + [
+                    "dump",
+                    "--max-value-size=0",
+                    "%s/_parameters/%s" % (run_id, task_id_params),
+                ]
+                step_cmds.extend(
+                    [
+                        "if ! %s >/dev/null 2>/dev/null; then %s; fi"
+                        % (" ".join(exists), " ".join(init))
+                    ]
+                )
+                input_paths = "%s/_parameters/%s" % (run_id, task_id_params)
+            elif (
+                node.type == "join"
+                and self.graph[node.split_parents[-1]].type == "foreach"
+            ):
+                # Set aggregated input-paths for a foreach-join
+                input_paths = (
+                    "$(python -m metaflow.plugins.argo.process_input_paths %s)"
+                    % input_paths
+                )
+            step = [
+                "step",
+                node.name,
+                "--run-id %s" % run_id,
+                "--task-id %s" % task_id,
+                "--retry-count %s" % retry_count,
+                "--max-user-code-retries %d" % user_code_retries,
+                "--input-paths %s" % input_paths,
+            ]
+            if any(self.graph[n].type == "foreach" for n in node.in_funcs):
+                # Pass split-index to a foreach task
+                step.append("--split-index {{inputs.parameters.split-index}}")
+            if self.tags:
+                step.extend("--tag %s" % tag for tag in self.tags)
+            if self.namespace is not None:
+                step.append("--namespace=%s" % self.namespace)
+
+            step_cmds.extend([" ".join(entrypoint + top_level + step)])
+
+            cmd_str = "%s; c=$?; %s; exit $c" % (
+                " && ".join([init_cmds, bash_capture_logs(" && ".join(step_cmds))]),
+                BASH_SAVE_LOGS,
+            )
+            cmds = shlex.split('bash -c "%s"' % cmd_str)
+
+            # Resolve resource requirements.
+            # TODO: Fix compute_resource_attributes def thats strewn everywhere
```

Comment:
Does this mean that argo doesn't use the same logic as `@kubernetes` and `@batch` now when dealing with resources? Or does it take resource attrs from `@kubernetes` _after_ compute resource attributes logic runs?

Reply:
The later. That change is in-flight.